### PR TITLE
DB API refresh part 2

### PR DESCRIPTION
**Pull request by vadimberezniker**
This PR updates most of the remaining call sites.

**Related issues**: N/A

### CODE REVIEW

**Review by tylerwilliams — APPROVED** [review]

File: enterprise/server/crypter_service/crypter_service.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 	"errors"
 	"flag"
 	"fmt"
+	"github.com/buildbuddy-io/buildbuddy/server/util/db"
```
Comment (bduffany) [inline enterprise/server/crypter_service/crypter_service.go:11]: nit: move down

File: enterprise/server/iprules/iprules.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package iprules
 import (
 	"context"
 	"flag"
+	"github.com/buildbuddy-io/buildbuddy/server/util/db"
```
Comment (bduffany) [inline enterprise/server/iprules/iprules.go:6]: here too

File: enterprise/server/usage/usage_test.go [hunk 3]
```diff
@@ -3,6 +3,7 @@ package usage_test
 import (
 	"context"
 	"fmt"
+	"github.com/buildbuddy-io/buildbuddy/server/util/db"
```
Comment (bduffany) [inline enterprise/server/usage/usage_test.go:6]: here as well

File: server/backends/invocationdb/invocationdb.go [hunk 4]
```diff
@@ -3,6 +3,7 @@ package invocationdb
 import (
 	"context"
 	"errors"
+	"github.com/buildbuddy-io/buildbuddy/server/util/db"
```
Comment (bduffany) [inline server/backends/invocationdb/invocationdb.go:6]: this line, in addition

File: server/util/db/db.go [hunk 5]
```diff
@@ -1135,3 +1140,29 @@ func ScanRows[T any](rq interfaces.DBRawQuery, fn func(ctx context.Context, val
...
+// slice of the specified type.
+//
+// This should not be used if you expect the result set size to be large.
+//
+// Example:
+//
+//	foos, err := db.All(rq, &tables.Foo{})
+//
+// TODO(vadim): check if there are any ScanRows calls that can be converted to ScanAll.
+func ScanAll[T any](rq interfaces.DBRawQuery, t *T) ([]*T, error) {
```
Comment (bduffany) [inline server/util/db/db.go:1154]: imo `ScanRows` vs. `ScanAll` is a bit unclear - maybe something like `CollectRows` or `ScanToSlice` could make sense?

**Review by bduffany — APPROVED** [review]

File: server/util/db/db.go (hunk 5 above)
Comment (vadimberezniker) [reply server/util/db/db.go:1154]: I renamed ScanRows to ScanEach. Hope that differentiates it enough.

**Pull request merged**